src/world/area_kmr/kmr_20/8EDFC0.c

```diff
@@ -146,8 +146,35 @@ void func_80242A04_8EE824(void) {
     queue_render_task(&rt);
 }
 
-void func_80242A3C_8EE85C(UnkKmr20*);
-INCLUDE_ASM(s32, "world/area_kmr/kmr_20/8EDFC0", func_80242A3C_8EE85C);
+void func_80242A3C_8EE85C(UnkKmr20* arg0) {
+    PlayerData* playerData = &gPlayerData;
+    s32 count;
+    s32 i;
+
+    count = 0;
+    for (i = 0; i < ARRAY_COUNT(playerData->badges); i++) {
+        if (playerData->badges[i] != 0) {
+            count++;
+        }
+    }
+    arg0->equippedBadges = count;
+
+    count = 0;
+    for (i = 0; ; i++) {
```

Comment:
Better to make this cond explicit I think
```suggestion
    for (i = 0; TRUE; i++) {
```